.github/actions/cache/action.yml

```diff
@@ -25,6 +28,16 @@ runs:
         echo "Current workflow path is: ${MIDDLE}"
         echo "path=${MIDDLE}" >> "$GITHUB_OUTPUT"
       shell: bash
+    - name: Generate meson cache key
+      if: ${{ inputs.use-meson == 'true' }}
+      run: |
+        echo "MESON_KEY=`meson --version`" >> $GITHUB_ENV
+      shell: bash
+    - name: Generate empty meson cache key
+      if: ${{ inputs.use-meson != 'true' }}
+      run: |
+        echo "MESON_KEY=\"none\"" >> $GITHUB_ENV
```

Comment:
Nit: Are the quotes around `none` necessary?

In line 34, we don't quote the output of `meson --version`.

Also, it would be more readable to include `meson-` in the value of `MESON_KEY`:

```
        echo "MESON_KEY=\"meson-none\"" >> $GITHUB_ENV
```